pkg/loop/loop.go

```diff
@@ -55,18 +55,18 @@ func New(options ...Option) *Loop {
 	return loop
 }
 
-// sleepUntilCtxDone return error when ctx is done during waiting
-func sleepUntilCtxDone(d time.Duration, ctx context.Context) error {
+// sleepUntilCtxDone sleep d duration until ctx canceled
```

Comment:
`until ctx done`

done maybe triggered by context timeout of deadline exceeded.